.github/workflows/CI.yml

```diff
@@ -117,6 +117,26 @@ jobs:
           command: fmt
           args: --all -- --check
 
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    container:
+      image: ubuntu:20.10
+    steps:
+      - run: apt-get update -y
+      - run: apt-get install -y libgtk-3-dev libglib2.0-dev libgraphene-1.0-dev git xvfb curl libcairo-gobject2 libcairo2-dev
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - run: rustup component add clippy
+      - uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: -- -D warnings
```

Comment:
For GStreamer I also do `-A clippy::redundant_pattern_matching -A clippy::single_match -A clippy::cast_lossless -A clippy::missing_safety_doc`, but that might just be a question of style :)

Reply:
I have added those per crate depending on what's used in auto, unless you prefer otherwise?

Reply:
Sure, that also works :)

Reply:
`#[allow(clippy::missing_safety_doc)]` I would add directly in `lib.rs` for everything though. These are bindings, everything is unsafe :P

Reply:
The thing is that a lot of glib is using that per function, not sure whether to remove all of those & set it on lib.rs for all the crates or not?

Reply:
Especially that for example, gtk has only one exported unsafe function, maybe it should be done per crate?

Reply:
I went ahead and handled that per crate. For glib, gdk, cairo (the three crates that exports a lot of unsafe functions) has it set in lib.rs & for others it's set per function

Reply:
Works for me